diff --git a/Makefile b/Makefile
index a11fc8d..266621a 100644
--- a/Makefile
+++ b/Makefile
@@ -6,10 +6,9 @@ INSTALLDIR?=$(DESTDIR)$(PREFIX)
 DOCDIR?=$(INSTALLDIR)/share/uzbl/docs
 RUN_PREFIX?=$(PREFIX)
 
-# use GTK3-based webkit when it is available
-USE_GTK3 = $(shell pkg-config --exists gtk+-3.0 webkitgtk-3.0 && echo 1)
+ENABLE_GTK3 ?= no
 
-ifeq ($(USE_GTK3),1)
+ifeq ($(ENABLE_GTK3),yes)
 	REQ_PKGS += gtk+-3.0 webkitgtk-3.0 javascriptcoregtk-3.0
 	CPPFLAGS = -DG_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED
 else
