--- cross-gnu	2013-04-07 22:15:27.880000000 +0200
+++ incubator-20130326-cross-gnu.patch.sh	2013-04-07 23:46:27.628000000 +0200
@@ -85,6 +85,8 @@
   "$SYS_ROOT""$CROSS_GNU_USR"/lib \
   "$TARGET"/ &&
 
+mkdir "$SYS_ROOT"/include &&
+
 
 # Install the cross GNU Binutils.
 
@@ -424,6 +426,21 @@
 fi &&
 
 
+# Install GNU Mach
+
+cd "$GNUMACH_OBJ"/ &&
+if ./config.status --version > /dev/null 2>&1; then :; else
+  # `$TARGET-gcc' doesn't work yet (to satisfy the Autoconf checks), but isn't
+  # needed either.
+  CC=gcc \
+  "$GNUMACH_SRC"/configure \
+    --host="$TARGET" \
+    --prefix="$CROSS_GNU_USR"
+fi &&
+"$MAKE" \
+  DESTDIR="$SYS_ROOT" \
+  install &&
+
 # Success.
 
 echo "$0"': Everything should be in place now.'
